lib/analyzer/image-inspector.ts

```diff
@@ -152,17 +155,166 @@ async function pullImage(
       platform,
     );
     if (pullAndSaveSuccessful) {
-      return;
+      imageNames.push(targetImage);
+      return imageNames;
     }
   }
 
-  await pullFromContainerRegistry(
+  const { indexDigest, manifestDigest } = await pullFromContainerRegistry(
     docker,
     targetImage,
     imageSavePath,
     username,
     password,
   );
+
+  const imageNameWithIndexDigest = getImageNameWithIndexDigest(
+    indexDigest,
+    targetImage,
+  );
+  const imageNameWithManifestDigest = getImageNameWithManifestDigest(
+    manifestDigest,
+    targetImage,
+  );
+
+  return getImageNames(
+    imageNameWithIndexDigest,
+    imageNameWithManifestDigest,
+    targetImage,
+  );
+}
+
+function getImageNames(
+  indexImageName: string | undefined,
+  manifestImageName: string | undefined,
+  targetImage: string,
+): string[] {
+  const imageNames: string[] = [];
```

Comment:
Nit: to ensure uniqueness, we may want to add these to a Set<string> (which will duplicate automatically), then return the resulting set as a string[]

Reply:
Sure, I can do that. I guess the `undefined` wouldn't get passed if I keep the if statements.